includes/service-providers/campaign_monitor/class-newspack-newsletters-campaign-monitor-usage-reports.php

```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * Campaign Monitor ESP Usage Report class.
+ *
+ * @package Newspack
+ */
+
+// The response from the API doesn't follow this rule.
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Campaign Monitor ESP Usage Report class
+ */
+class Newspack_Newsletters_Campaign_Monitor_Usage_Reports {
+
+	/**
+	 * Gets the Campaign Monitor Lists API client.
+	 *
+	 * @param string $list_id List ID.
+	 * @return CS_REST_Lists|WP_Error CS_REST_Lists instance or error.
+	 */
+	private static function get_lists_client( $list_id ) {
+		$cm      = Newspack_Newsletters_Campaign_Monitor::instance();
+		$api_key = $cm->api_key();
+
+		if ( ! $api_key ) {
+			return new WP_Error(
+				'newspack_newsletters_missing_api_key',
+				__( 'No Campaign Monitor API key available.', 'newspack-newsletters' )
+			);
+		}
+
+		return new CS_REST_Lists( $list_id, [ 'api_key' => $api_key ] );
+	}
+
+	/**
+	 * Gets the Campaign Monitor Clients API client.
+	 *
+	 * @return CS_REST_Clients|WP_Error CS_REST_Clients instance or error.
+	 */
+	private static function get_clients_client() {
+		$cm        = Newspack_Newsletters_Campaign_Monitor::instance();
+		$api_key   = $cm->api_key();
+		$client_id = $cm->client_id();
+
+		if ( ! $api_key ) {
+			return new WP_Error(
+				'newspack_newsletters_missing_api_key',
+				__( 'No Campaign Monitor API key available.', 'newspack-newsletters' )
+			);
+		}
+		if ( ! $client_id ) {
+			return new WP_Error(
+				'newspack_newsletters_missing_client_id',
+				__( 'No Campaign Monitor Client ID available.', 'newspack-newsletters' )
+			);
+		}
+
+		return new CS_REST_Clients( $client_id, [ 'api_key' => $api_key ] );
+	}
+
+	/**
+	 * Gets the full report
+	 *
+	 * @return Newspack_Newsletters_Service_Provider_Usage_Report
+	 */
+	public static function get_report() {
+		$subs      = self::get_subscribers_info();
+		$campaigns = self::get_campaigns_info();
+		if ( is_wp_error( $subs ) ) {
+			return $subs;
+		}
+		if ( is_wp_error( $campaigns ) ) {
+			return $campaigns;
+		}
+		return new Newspack_Newsletters_Service_Provider_Usage_Report( array_merge( $subs, $campaigns ) );
+	}
+
+	/**
+	 * Get subscribers and unsubscribers for the last $days days.
+	 *
+	 * @return array|WP_Error Array of subscribers and unsubscribers for the last $days days, or WP_Error on failure.
```

Comment:
Function comment mentions `$days`, but it's not in the function body.

Reply:
thanks

Reply:
wow and there's a trailing comma in the arguments list: `get_subscribers_for_list( $list_id, ) {`

fixing it now